packages/path_provider/path_provider/lib/path_provider.dart

```diff
@@ -3,14 +3,36 @@
 // found in the LICENSE file.
 
 import 'dart:async';
-import 'dart:io' show Directory;
+import 'dart:io' show Directory, Platform;
 
+import 'package:flutter/foundation.dart' show visibleForTesting;
+import 'package:path_provider_linux/path_provider_linux.dart';
 import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
 
 export 'package:path_provider_platform_interface/path_provider_platform_interface.dart'
     show StorageDirectory;
 
-PathProviderPlatform get _platform => PathProviderPlatform.instance;
+// Should not be used in users' code, just made visible for testing to test the method channel implementation
+@visibleForTesting
+set testOverrideUseMethodChannel(bool override) {
+  _useMethodChannel = override;
+}
+
+bool _useMethodChannel = false;
+PathProviderPlatform __platform;
+
+// This is to manually endorse the linux path provider until automatic registration of dart plugins is implemented.
```

Comment:
Per [my comment here](https://github.com/flutter/plugins/pull/2631#discussion_r406489237) I'm not sure if we want this hidden; there are legitimate reasons some could want to call this.

@tvolkert Who is the right person to weigh in on this at the moment with amirh OOO?

Reply:
Renamed the platform override and made it user visible in my latest commit, I doubt that people will be using their own registered version of `path_provider` unless they are using a different embedding i.e. `go-flutter`, but it will mean that those implementations will have to add `disablePathProviderPlatformOverride = true` until we have automatic registration.

Reply:
@stuartmorgan 
I've made this user visible, and renamed / reformatted it some. Should we continue waiting for a reply? The only issue would be that removing it in the future would be technically a breaking change for anyone who is using it.

Reply:
I was thinking about this last night, and was going to updating this morning actually. Since getting resolution on this is proving difficult, let's return it to being test-only. That means we're not adding to the public API surface, so it won't need the same level of discussion/approval. We can always add it later if it becomes necessary, without it blocking getting the initial implementation landed.